hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -229,7 +241,8 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
     tableLevelMutex.lock();
     try {
       lockId = Optional.of(acquireLock());
-      // TODO add lock heart beating for cases where default lock timeout is too low.
+      hiveLockHeartbeat = new HiveLockHeartbeat(metaClients, lockId.get(), lockCheckMaxWaitTime);
```

Comment:
This is how Hive schedules the heartbeat:
```
HiveConf.getTimeVar(conf, HiveConf.ConfVars.HIVE_TXN_TIMEOUT, TimeUnit.MILLISECONDS) / 2;
```

Reply:
Yea, but Iceberg does not have this config, so we can't use that. The reason I used `iceberg.hive.lock-check-max-wait-ms` is that we already advise users to keep this lower than `hive.txn.timeout`. I would be open to making this even smaller to something like `lockCheckMaxWaitTime / 2` to allow room for delays in heartbeats and/ or retries. Thoughts @pvary ?

Reply:
I would prefer a different config for the heartbeat period